app/src/main/java/tech/ula/ui/AppListFragment.kt

```diff
@@ -52,49 +63,58 @@ class AppListFragment : Fragment() {
 
     override fun onActivityCreated(savedInstanceState: Bundle?) {
         super.onActivityCreated(savedInstanceState)
-
-        appListViewModel.getAllApps().observe(viewLifecycleOwner, appChangeObserver)
-
         activityContext = activity!!
-
+        appListViewModel.getAllApps().observe(viewLifecycleOwner, appChangeObserver)
         registerForContextMenu(list_apps)
         list_apps.onItemClickListener = AdapterView.OnItemClickListener {
             _, _, position, _ ->
             val selectedApp = appList[position]
-            println("Clicked on APP: ${selectedApp.name}")
+
+            // TODO: Only show if not installed
+            showAppDetails(app = selectedApp)
+        }
+
+        val swipeLayout = activityContext.findViewById<SwipeRefreshLayout>(R.id.swipe_refresh)
+        swipeLayout.setOnRefreshListener(
+                SwipeRefreshLayout.OnRefreshListener {
+                    appListViewModel.refreshAppsList()
+                    while (appListViewModel.getRefreshStatus() == RefreshStatus.ACTIVE) {
+                        Thread.sleep(2000)
```

Comment:
Decrease this to 500 to match other polls. We should revisit lowering it even further in the future to increase responsiveness.